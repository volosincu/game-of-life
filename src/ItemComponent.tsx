// @ts-nocheck
import {store} from "./store";
import {renderRating} from "./helpers";
import React, { useContext } from 'react';
import { useInView } from 'react-intersection-observer'


export const ItemComponent = React.memo((props: ItemProps) => {
    const globalState = useContext(store);
    const { dispatch } = globalState;
    const { imgsrc, name, price, rating, room} = props;
    
    const [ref, inView, entry] = useInView({
        //triggerOnce: true
    });

    return <li className="hotel" ref={ref}>
        <img
          src={inView ? imgsrc : ""}
          className="img-l"
          alt={name}
          loading="lazy"
        />
        <div className="hotel-info">
            <label className="hotel-name">{name}</label>
            <div>{renderRating(rating)}</div>
            <label className="hotel-room-type">Room Type: <span>{room}</span></label>
        </div>
        <div className="hotel-menu-l">
            <label className="hotel-price">{price}$</label>
        </div>
    </li>
});
