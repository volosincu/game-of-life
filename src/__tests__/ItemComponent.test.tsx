// @ts-nocheck
import React from 'react';
import { shallow } from "enzyme";
import { ItemComponent } from '../ItemComponent';


describe("ItemComponent component", ()=> {
    let wrapper: any;
    beforeEach(()=> {
        const props = {
            price: 200,
            name: "Hotel Regina Maria",
            rating: 5.0,
            room: "Studio Apartament",
            imgsrc: "https://hotelimages.webjet.com.au/hotels/7000000/6430000/6428700/6428660/215090a4_z.jpg"
        };
        wrapper = shallow(<ItemComponent key="1" {...props} />);
    });

    it('renders correctly - toMatchSnapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
