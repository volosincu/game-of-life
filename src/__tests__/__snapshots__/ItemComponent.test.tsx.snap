// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemComponent component renders correctly - toMatchSnapshot 1`] = `ShallowWrapper {}`;
